drop-boxes/register-raw-and-mzML/register-raw-and-mzml.py

```diff
@@ -0,0 +1,269 @@
+"""
+This etl script registers a pair consisting of a proteomics raw file or folder and its converted mzML and registers
+both to the same openBIS sample. Some metadata is parsed from the mzML and stored.
+
+Incoming raw files (of different vendor formats) are supported.
+
+This script reuses some logic from the raw file conversion etl script
+
+The stdout of this file is redirected to
+`~openbis/servers/datastore_server/log/startup_log.txt`
+"""
+
+import sys
+sys.path.append('/home-link/qeana10/bin/')
+import os
+import time
+import re
+import subprocess
+import datetime
+import xml.etree.ElementTree
+from functools import partial
+import ch.systemsx.cisd.etlserver.registrator.api.v2
+from ch.systemsx.cisd.openbis.generic.shared.api.v1.dto import (
+    SearchCriteria, SearchSubCriteria
+)
+######## Sample Tracking related import
+from life.qbic.sampletracking import SampleTracker
+from life.qbic.sampletracking import ServiceCredentials
+from java.net import URL
+
+import sample_tracking_helper_qbic as tracking_helper
+#### Setup Sample Tracking service
+SERVICE_CREDENTIALS = ServiceCredentials()
+SERVICE_CREDENTIALS.user = tracking_helper.get_service_user()
+SERVICE_CREDENTIALS.password = tracking_helper.get_service_password()
+SERVICE_REGISTRY_URL = URL(tracking_helper.get_service_reg_url())
+DATA_AVAILABLE_JSON = tracking_helper.get_data_available_status_json()
+
+### We need this object to update the sample status later
+SAMPLE_TRACKER = SampleTracker.createLocationIndependentSampleTracker(SERVICE_REGISTRY_URL, SERVICE_CREDENTIALS)
+
+# *Q[Project Code]^4[Sample No.]^3[Sample Type][Checksum]*.*
+barcode_pattern = re.compile('Q[a-zA-Z0-9]{4}[0-9]{3}[A-Z][a-zA-Z0-9]')
+ms_pattern = re.compile('MS[0-9]*Q[A-Z0-9]{4}[0-9]{3}[A-Z][A-Z0-9]')
+ms_prefix_pattern = re.compile('MS[0-9]*')
+
+VENDOR_FORMAT_EXTENSIONS = {'.raw':'RAW_THERMO', '.d':'D_BRUKER','.wiff':'WIFF_SCIEX'}
+WATERS_FORMAT = "RAW_WATERS"
+
+def extract_barcode(filename):
+    """Extract valid barcodes from the filename.
+
+    Return project_id, experiment_id and the whole barcode.
+    If no barcode was found, raise a ValueError. If the file
+    contains a qbic barcode with an invalid checksum, raise
+    a RuntimeError.
+
+    TODO rewrite
+    """
+    try:
+        code = barcode[0:-1]
+        return checksum.checksum(code) == barcode[-1]
+    except:
+        return True
+
+def parse_timestamp_easy(mzml_path):
+    with open(mzml_path, 'r') as mzml:
+        time = None
+        for line in mzml:
+            if "<run id=" in line:
+                for token in line.split(" "):
+                    if "startTimeStamp=" in token:
+                        xsdDateTime = token.split('"')[1]
+                        time = datetime.datetime.strptime(xsdDateTime, '%Y-%m-%dT%H:%M:%SZ').strftime('%Y-%m-%d %H:%M:%S')
+                break
+        return time
```

Comment:
Out of interest, is this run id the one provided by nextflow tower(since we shut that down no run id will be generated)

Reply:
no, this is information generated by the mass spectrometer